server/util/signalutil/signalutil.go

```diff
@@ -0,0 +1,19 @@
+package signalutil
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+func OnSignal(s syscall.Signal, fn func() error, errFn func(error)) {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, s)
+	go func() {
+		for range c {
+			if err := fn(); err != nil {
+				errFn(err)
+			}
+		}
+	}()
+}
```

Comment:
I am a bit concerned about this func obscuring the fact that a goroutine is created and never cleaned up, which could lead to a goroutine leak if this is used improperly.

I think we should avoid introducing this util and instead use `signal.Notify() / go func() ...` directly.

Reply:
directly in `ReloadOnSIGHUP`? or directly in the `main` functions after calling `PopulateFlagsFromFile`? If the former, I suppose we could cleanup a previous goroutine if it exists and then start the goroutine, would that be good?

Reply:
Could also name it something like "MakeSignalListener" (likely terrible name, but that idea) to emphasize that when you make one, it sticks around and does something?

Reply:
I think it'd be fine to do it in ReloadOnSIGHUP and just add a clarifying comment to ReloadOnSIGHUP that this is only intended to be called in `main`.

Reply:
Cool; sounds great.